corehq/messaging/fcm/utils.py

```diff
@@ -70,3 +71,44 @@ def send_to_multiple_devices(self, registration_tokens, title='', body='', data=
         )
         response = messaging.send_multicast(message, app=self.app)
         return response
+
+
+def parse_multicast_batch_response(multicast_response, registration_tokens):
+    """
+    Parses the multicast batch response to a json format as below for easier usage.
+    {
+        'success_count': <integer>
+        'failure_count': <integer>
+        'success_responses':[
+            {
+                'token': <fcm_device_token>
+                'message_id': <message_id>
+            }
+        ]
+        'failure_responses':[
+            {
+                'token': <fcm_device_token>
+                'exception':<fcm exception that occurred>
+            }
+        ]
+    }
+    'registration_tokens' parameter list have same order as the one sent to 'send_to_multiple_devices'
+    """
+    parsed_response = {
+        'success_count': multicast_response.success_count,
+        'failure_count': multicast_response.failure_count,
+        'success_responses': [],
+        'failure_responses': []
+    }
+    for idx, resp in enumerate(multicast_response.responses):
+        if resp.success:
+            parsed_response['success_responses'].append({
+                'token': registration_tokens[idx],
```

Comment:
I don't think we want to store the tokens in the logs since its a sensitive information about the device.

Reply:
Since we are storing this only when all devices fail, won't that mean that success_count would be 0 and there would not be anything in success_responses?

Reply:
> okay, thinking about this more, this is a device token which does not let someone gain access to device in case its leaked, and is just a way to identify the device so it should be okay for it to be in the logs. Can you confirm this with Shubham once?

Yes this can be used only for sending our FCM notification which again will require server certificate access to do so !

> I believe there is only one use where we are parsing only if all responses are failures?
Yes that is correct, however the method is developed to be reusable in future oif they wish to parse the FCM batch response as JSON.

Reply:
This is addressed in  0fc90783a86c6d79aa139ba242cacb6204e82cd2